alphagov/paas-cf
# [#110652158] Cf upgrade

## What

Make everything newerer and betterer

* Stemcell version for concourse-deployer and bosh upgraded to latest (3197) to resolve libc vulnerability
* Stemell version for CF also updated  to latest version (3197)
* CF to version 228 (allowing us some future versions to practice upgrading again)
* Diego to 1450 (matches cf228 and again leave us future versions for upgrading)
* Etcd to version 27 (to match diego version 1450)
* Garden-linux to 332 (to match diego version 1450)
* Import manifest changes from CF228 release into our own manifest stubs

## How to test
Due to the postgres ccdb role being deployed as a single instance, it is not possible to upgrade an existing deployment (the postgress role cannot be stopped, so the disk cannot be unmounted) - Because of this, you will need to do a clean deployment to test. 

* Review the commit history to see what has been changed.
* Deploy an environment as normal by following the instructions in the `README.md` 
* Confirm deployment passes smoke-tests and custom-acceptance-tests pass successfully
* There is around a 75% chance that the CF acceptance-tests will pass too (for extant reasons - some kind of system-load/race-condition issue) - We have seen these tests pass three out of four times when working on this story.
* Run `conccourse/scripts/bosh-init.sh <env>` to gain access to the bosh CLI, and check 'bosh stemcells' and 'bosh deployments' to confirm deployed versions
* Check everything works as you'd expect it to.

## Who can review

Anyone on the team except @alext or myself
